crates/astria-core/src/sequencer/v1alpha1/transaction/action.rs

```diff
@@ -975,3 +1012,198 @@ enum FeeAssetChangeActionErrorKind {
     #[error("the asset_id was missing")]
     MissingAssetId,
 }
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone)]
+pub struct InitBridgeAccountAction {
+    pub rollup_id: RollupId,
+    pub asset_ids: Vec<asset::Id>,
+    pub fee_asset_id: asset::Id,
+}
+
+impl InitBridgeAccountAction {
+    #[must_use]
+    pub fn into_raw(self) -> raw::InitBridgeAccountAction {
+        raw::InitBridgeAccountAction {
+            rollup_id: self.rollup_id.to_vec(),
+            asset_ids: self
+                .asset_ids
+                .iter()
+                .map(|id| id.as_bytes().to_vec())
+                .collect(),
+            fee_asset_id: self.fee_asset_id.as_bytes().to_vec(),
+        }
+    }
+
+    #[must_use]
+    pub fn to_raw(&self) -> raw::InitBridgeAccountAction {
+        raw::InitBridgeAccountAction {
+            rollup_id: self.rollup_id.to_vec(),
+            asset_ids: self
+                .asset_ids
+                .iter()
+                .map(|id| id.as_bytes().to_vec())
+                .collect(),
+            fee_asset_id: self.fee_asset_id.as_bytes().to_vec(),
+        }
+    }
+
+    /// Convert from a raw, unchecked protobuf [`raw::InitBridgeAccountAction`].
+    ///
+    /// # Errors
+    ///
+    /// - if the `rollup_id` field is invalid
+    pub fn try_from_raw(
+        proto: raw::InitBridgeAccountAction,
+    ) -> Result<Self, InitBridgeAccountActionError> {
+        let rollup_id = RollupId::try_from_slice(&proto.rollup_id)
+            .map_err(InitBridgeAccountActionError::invalid_rollup_id)?;
+        let asset_ids = proto
+            .asset_ids
+            .into_iter()
+            .map(|bytes| asset::Id::try_from_slice(&bytes))
+            .collect::<Result<Vec<asset::Id>, asset::IncorrectAssetIdLength>>()
+            .map_err(InitBridgeAccountActionError::invalid_asset_id)?;
+        let fee_asset_id = asset::Id::try_from_slice(&proto.fee_asset_id)
+            .map_err(InitBridgeAccountActionError::invalid_fee_asset_id)?;
+
+        Ok(Self {
+            rollup_id,
+            asset_ids,
+            fee_asset_id,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct InitBridgeAccountActionError(InitBridgeAccountActionErrorKind);
+
+impl InitBridgeAccountActionError {
+    #[must_use]
+    fn invalid_rollup_id(err: IncorrectRollupIdLength) -> Self {
+        Self(InitBridgeAccountActionErrorKind::InvalidRollupId(err))
+    }
+
+    #[must_use]
+    fn invalid_asset_id(err: asset::IncorrectAssetIdLength) -> Self {
+        Self(InitBridgeAccountActionErrorKind::InvalidAssetId(err))
+    }
+
+    #[must_use]
+    fn invalid_fee_asset_id(err: asset::IncorrectAssetIdLength) -> Self {
+        Self(InitBridgeAccountActionErrorKind::InvalidFeeAssetId(err))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[allow(clippy::enum_variant_names)]
```

Comment:
I think the `allow` here is unnecessary. Just remove the `Invalid*` prefixes.

In general, please add explainers why we deviate from the pedantic clippy. It's ok for the `*Action` types because that's what the entire crate does (to match the protos), but otherwise I wouldn't do it

Reply:
i put the `Invalid` prefix for consistency for other errors, just removed it + the clippy allow

Reply:
ok nvm, reverted it back and added a comment. clippy is angry still even if i remove the prefix: `error: all variants have the same postfix: `Id`` but removing the postfix makes no sense